base/applications/regedit/framewnd.c

```diff
@@ -807,6 +814,98 @@ static void ChooseFavorite(LPCWSTR pszFavorite)
         RegCloseKey(hKey);
 }
 
+static LPWSTR GetItemFullPath(HTREEITEM hTI)
+{
+    HKEY hRoot;
+    WCHAR rootname[MAX_PATH], *buffer;
+    SIZE_T rootlen, subkeylen;
+    LPCWSTR subkey = GetItemPath(g_pChildWnd->hTreeWnd, hTI, &hRoot);
+    if (!subkey || !hRoot)
+        return NULL;
+    if (!GetKeyName(rootname, ARRAY_SIZE(rootname), hRoot, L""))
+        return NULL;
+    rootlen = lstrlenW(rootname) + 1; // + 1 for '\\'
+    subkeylen = lstrlenW(subkey);
+    buffer = (WCHAR*)GlobalAlloc(GMEM_FIXED, (rootlen + subkeylen + 1) * sizeof(WCHAR));
+    if (buffer)
+    {
+        lstrcpyW(buffer, rootname);
+        buffer[rootlen - 1] = '\\';
+        lstrcpyW(buffer + rootlen, subkey);
+    }
+    return buffer;
+}
+
+static INT_PTR CALLBACK AddToFavoritesDlgProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+    HWND hName = GetDlgItem(hWnd, IDC_FAVORITENAME);
+    WCHAR name[MAX_PATH];
+    switch (uMsg)
+    {
+        case WM_INITDIALOG:
+        {
+            TVITEM tvi;
+            tvi.mask = TVIF_HANDLE | TVIF_TEXT;
+            tvi.hItem = TreeView_GetSelection(g_pChildWnd->hTreeWnd);
+            tvi.pszText = name;
+            tvi.cchTextMax = _countof(name);
+            if (!TreeView_GetItem(g_pChildWnd->hTreeWnd, &tvi))
+                tvi.pszText[0] = UNICODE_NULL;
+            SendMessageW(hName, WM_SETTEXT, 0, (LPARAM)tvi.pszText);
+            SendMessageW(hName, EM_LIMITTEXT, _countof(name) - 1, 0);
+            return TRUE;
+        }
+        case WM_COMMAND:
+            switch (LOWORD(wParam))
+            {
+                case IDOK:
+                    {
+                        LPCWSTR path;
+                        HKEY hKey;
+                        DWORD err;
+                        if (GetWindowTextW(hName, name, _countof(name)))
+                            err = ERROR_SUCCESS;
+                        else
+                            err = GetLastError();
+
+                        if (err == ERROR_SUCCESS)
+                        {
+                            path = GetItemFullPath(NULL);
+                            if (!path)
+                                err = ERROR_NOT_ENOUGH_MEMORY;
+                        }
+
+                        if (err == ERROR_SUCCESS)
+                            err = RegCreateKeyExW(HKEY_CURRENT_USER, s_szFavoritesRegKey, 0,
+                                                  NULL, 0, KEY_SET_VALUE, NULL, &hKey, NULL);
```

Comment:
If above suggestion adopted, then also,
```suggestion
                        err = RegCreateKeyExW(HKEY_CURRENT_USER, s_szFavoritesRegKey, 0,
                                              NULL, 0, KEY_SET_VALUE, NULL, &hKey, NULL);
```